modules/client/src/main/java/org/apache/ignite/internal/client/sql/ClientSql.java

```diff
@@ -264,15 +258,57 @@ public <T> CompletableFuture<AsyncResultSet<T>> executeAsync(
     /** {@inheritDoc} */
     @Override
     public CompletableFuture<long[]> executeBatchAsync(@Nullable Transaction transaction, String query, BatchedArguments batch) {
-        // TODO IGNITE-17059.
-        throw new UnsupportedOperationException("Not implemented yet.");
+        return executeBatchAsync(transaction, new StatementImpl(query), batch);
     }
 
     /** {@inheritDoc} */
     @Override
     public CompletableFuture<long[]> executeBatchAsync(@Nullable Transaction transaction, Statement statement, BatchedArguments batch) {
-        // TODO IGNITE-17059.
-        throw new UnsupportedOperationException("Not implemented yet.");
+        PayloadWriter payloadWriter = w -> {
+            writeTx(transaction, w);
+
+            w.out().packString(statement.defaultSchema());
+            w.out().packInt(statement.pageSize());
+            w.out().packLong(statement.queryTimeout(TimeUnit.MILLISECONDS));
+            w.out().packNil(); // sessionTimeout
+            w.out().packString(statement.timeZoneId().getId());
+
+            packProperties(w, null);
+
+            w.out().packString(statement.query());
+            w.out().packBatchedArgumentsAsBinaryTupleArray(batch);
+            w.out().packLong(ch.observableTimestamp());
+        };
+
+        PayloadReader<BatchResultInternal> payloadReader = r -> {
+            ClientMessageUnpacker unpacker = r.in();
+
+            unpacker.skipValues(4); // skipping values that are not currently in use.
+
+            long[] updateCounters = unpacker.unpackLongArray();
+
+            if (unpacker.tryUnpackNil()) {
+                unpacker.skipValues(2);
```

Comment:
Please add a comment explaining what is skipped an why.

Reply:
Done.